gpMgmt/bin/go-tools/cli/signal_test.go

```diff
@@ -0,0 +1,214 @@
+package cli_test
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/greenplum-db/gp-common-go-libs/testhelper"
+	"github.com/greenplum-db/gpdb/gp/cli"
+	"github.com/greenplum-db/gpdb/gp/testutils"
+)
+
+func TestHandleSignal(t *testing.T) {
+	_, _, logfile := testhelper.SetupTestLogger()
+
+	// initialize the states based on the stream controller implementation
+	const (
+		streamRunning = iota
+		streamPaused
+		streamDiscard
+	)
+
+	t.Run("prompts the user and correctly sets the TerminationRequested flag when the signal is SIGINT", func(t *testing.T) {
+		defer func() {
+			cli.TerminationRequested = false
+		}()
+
+		buffer, writer, resetStdout := testutils.CaptureStdout(t)
+		defer resetStdout()
+
+		// when user selects yes
+		resetStdin := testutils.MockStdin(t, "y\n")
+		defer resetStdin()
+
+		cli.HandleSignal(syscall.SIGINT, nil)
+		writer.Close()
+
+		testutils.AssertLogMessage(t, logfile, `\[WARNING\]:-received an interrupt signal`)
+
+		stdout := <-buffer
+		expectedPromptText := "Do you want to continue terminating the current execution?"
+		if !strings.Contains(stdout, expectedPromptText) {
+			t.Fatalf("got %s, want %s", stdout, expectedPromptText)
+		}
+
+		if !cli.TerminationRequested {
+			t.Fatalf("got %t, want true", cli.TerminationRequested)
+		}
+
+		// when user selects no
+		resetStdin = testutils.MockStdin(t, "n\n")
+		defer resetStdin()
+
+		cli.HandleSignal(syscall.SIGINT, nil)
+		if cli.TerminationRequested {
+			t.Fatalf("got %t, want false", cli.TerminationRequested)
+		}
+	})
+
+	t.Run("correctly sets the stream controller states when user selects yes", func(t *testing.T) {
+		defer func() {
+			cli.TerminationRequested = false
+		}()
+
+		writer, resetStdin := testutils.MockStdinWithWriter(t)
+		defer resetStdin()
+
+		ctrl := cli.NewStreamController()
+		go cli.HandleSignal(syscall.SIGINT, ctrl) // the code will be blocked until we indicate that the stream is paused using ctrl.SetPaused()
+
+		// the stream should be paused initially to display the prompt
+		time.Sleep(100 * time.Millisecond)
+		if ctrl.State() != streamPaused {
+			t.Fatalf("expected stream to be paused")
+		}
+
+		ctrl.SetPaused() // indicate that the stream has been paused so we can display the user prompt
+
+		// enter yes in the prompt
+		_, err := writer.WriteString("y\n")
```

Comment:
This will reduce the confusion about the exact input :
```suggestion
		_, err := writer.WriteString(fmt.Sprintln("y"))
```

Reply:
sure